pandas/core/common.py

```diff
@@ -1374,36 +1374,44 @@ def _fill_zeros(result, x, y, name, fill):
     mask the nan's from x
     """
 
-    if fill is not None:
+    if fill is None or is_float_dtype(result):
+        return result
+
+    if name.startswith(('r', '__r')):
+        x,y = y,x
 
-        if name.startswith('r'):
-            x,y = y,x
+    if not isinstance(y, np.ndarray):
+        dtype, value = _infer_dtype_from_scalar(y)
+        y = np.empty(result.shape, dtype=dtype)
```

Comment:
do you really need to do this? you end up using 2x memory then (broadcasts don't generally need to expand memory that much)